bin/setupReplication.js

```diff
@@ -0,0 +1,449 @@
+                }],
+            },
+        };
+        this._s3.source.putBucketReplication(params, (err, res) => {
+            if (err) {
+                this._log.error('error enabling replication', {
+                    method: '_SetupReplication._enableReplication',
+                    error: err.message,
+                    errStack: err.stack,
+                });
+                return cb(err);
+            }
+            this._log.debug('Bucket replication enabled', {
+                response: res,
+                method: '_enableReplication',
+            });
+            return cb(null, res);
+        });
+    }
+
+    _parallelTasks(cb) {
+        async.parallel({
+            sourceBucket: next => this._createBucket('source', next),
+            targetBucket: next => this._createBucket('target', next),
+            sourceRole: next => this._createRole('source', next),
+            targetRole: next => this._createRole('target', next),
+            sourcePolicy: next => this._createPolicy('source', next),
+            targetPolicy: next => this._createPolicy('target', next),
+        }, (err, res) => {
+            cb(err, res);
+        });
+    }
+
+    _seriesTasks(data, cb) {
+        const sourceRole = data.sourceRole.Role;
+        const targetRole = data.targetRole.Role;
+        const sourcePolicyArn = data.sourcePolicy.Policy.Arn;
+        const targetPolicyArn = data.targetPolicy.Policy.Arn;
+        const roleArns = `${sourceRole.Arn},${targetRole.Arn}`;
+
+        async.series([
+            next => this._enableVersioning('source', next),
+            next => this._enableVersioning('target', next),
+            next => this._attachResourcePolicy(sourcePolicyArn,
+                sourceRole.RoleName, 'source', next),
+            next => this._attachResourcePolicy(targetPolicyArn,
+                targetRole.RoleName, 'target', next),
+            next => this._enableReplication(roleArns, next),
+        ], (err, res) => {
+            cb(err, res);
+        });
+    }
+
+    run(cb) {
+        async.waterfall([
+            next => this._parallelTasks((err, setupInfo) => {
+                next(err, setupInfo);
+            }),
+            (setupInfo, next) => this._seriesTasks(setupInfo, (err, res) => {
+                next(err, res);
+            }),
+            (solo, next) => this._checkSanity((err, res) => {
+                next(err, res);
+            }),
+        ], cb);
+    }
+}
+
+commander
+  .version('0.1.0')
+  .arguments('<source> <destination>')
+  .action((source, destination) => {
+      const log = new Logger('BackbeatSetup').newRequestLogger();
+      const s = new _SetupReplication(source, destination, log, config);
+      s.run((err, res) => {
+          if (err) {
+              log.info(err);
+              return log.info('replication script failed');
+          }
+          log.info(res);
+          return log.info('replication setup successful');
```

Comment:
Probably one `log.info` with the message `'replication script failed'` would suffice, and then include the `res` in the object following the message, as you've done in the `log.error` calls.